### ISSUE TITLE
你好，我想问下这个目标检测任务中有做目标分类吗？我看这个展示的结果都没有目标种类。尝试了改一下代码中的设置也没有成功，请大佬赐教！

### ISSUE THREAD

**Issue by hddlovefxx** [hustvl/YOLOP#86]

**Comment by superXR** [comment]
输出结果里面有分类啊，自己对应画上去就行啊

**Comment by RuiwenZhang** [comment]
同问

**Comment by xinshoushangludd** [comment]
utils\plot.py 里面的函数 plot_one_box 那儿   将if label后面的 不要隐去 ，就行了

**Comment by superXR** [comment]
没有分类，他把{'car', 'bus', 'truck','train'} 这几类都作为一个类别去训练的，这一点在dataset/bdd.py 里面可以看到；想要分类需要重新训练，除了修改bdd.py，还要参照yolo去修改类别数量nc，有些地方他都是hardcode写死的，需要自己到代码里面去改，另外还要loss代码也要修改，只修改BCEcls，其中pos_weight=torch.ones([det_num_class])，自己可以试着调一调

**Comment by Yaoxingtian** [comment]
> pos_weight

你说的是这个吗？ BCEcls = nn.BCEWithLogitsLoss(pos_weight=torch.Tensor([cfg.LOSS.CLS_POS_WEIGHT])).to(device)，没有pos_weight=torch.ones([det_num_class])， 这里是权重的参数，没有det_num_class

**Comment by superXR** [comment]
> > pos_weight
> 
> 你说的是这个吗？ BCEcls = nn.BCEWithLogitsLoss(pos_weight=torch.Tensor([cfg.LOSS.CLS_POS_WEIGHT])).to(device)，没有pos_weight=torch.ones([det_num_class])， 这里是权重的参数，没有det_num_class

我说的这个det_num_class需要你自己写，代表要检测的类别的数量，也就是你数据训练时输入的类别数量，这样设置就还是正负样本权重相同，只不过要与你要检测的类别数量对应起来，只是修改了pos_weight的维度

**Comment by thinkthinking** [comment]
作者在github上的预训练模型包括代码默认都是只识别single class的，如果想要多类别识别，需要进行修改网络结构以及训练参数，重新训练。
